File: src/main/java/com/cryptox/service/WalletService.java
package com.cryptox.service;

import com.cryptox.dto.response.ApiResponse;
import com.cryptox.entity.User;
import com.cryptox.entity.Wallet;

import java.math.BigDecimal;

public interface WalletService {

    Wallet createWallet(User user);

    Wallet getWallet(Long userId);

    ApiResponse getMyWallet(String email);

    ApiResponse deposit(String email, BigDecimal amount);
}

File: frontend/src/services/walletService.js
import api from "./api";

export const getMyWallet = async () => {
    const response = await api.get("/wallet/me");
    return response.data;
};

export const depositMoney = async (amount) => {
    const response = await api.post("/wallet/deposit", {
        amount,
    });

    return response.data;
};

export const withdrawMoney = async (amount) => {
    const response = await api.post("/wallet/withdraw", {
        amount,
    });

    return response.data;
};